simapp/app.go

```diff
@@ -131,7 +131,7 @@ type SimApp struct {
 
 // NewSimApp returns a reference to an initialized SimApp.
 func NewSimApp(
```

Comment:
TBH, I'm not totally thrilled about updating the app's constructor this way simply to proxy the argument to the upgrade module's constructor. But I cannot think of a cleaner way of doing it :-/

Thoughts @aaronc @ethanfrey?

Reply:
Not blocking, but I think we can think of cleaner way by passing in keeper "option" functions similar to the way we handle baseapp. However, this requires more plumbing and most notably requires the semantics of keepers to chance by operating on references and not copies.

Reply:
I like your approach @alexanderbez